application/src/main/java/org/thingsboard/server/controller/UserController.java

```diff
@@ -378,4 +381,46 @@ public void setUserCredentialsEnabled(
         }
     }
 
+    @ApiOperation(value = "Save user settings (saveUserSettings)",
+            notes = "Save user settings for specified user id. " )
+    @PreAuthorize("hasAnyAuthority('SYS_ADMIN', 'TENANT_ADMIN', 'CUSTOMER_USER')")
+    @PostMapping(value = "/user/{userId}/settings")
```

Comment:
Remove `{userId}` from all three methods and use getCurrentUser to get the id and tenant id of the current user entity

Reply:
```java
    public JsonNode saveUserSettings(@RequestBody JsonNode userSettings) throws ThingsboardException;
```